Dynamic_Bar.lua

```diff
@@ -19,34 +17,28 @@ Type:UsesAttributes("value, maxValue, valueColor")
 Type:UsesAttributes("state")
 Type:UsesAttributes("texture")
 
-TMW:RegisterUpgrade(45013, {
-	icon = function(self, ics)
-		if ics.Type == "dynamicbar" then
-			ics.Alpha = 1
-			ics.UnAlpha = ics.ConditionAlpha or 0
-			ics.ConditionAlpha = 0
-		end
-	end,
+local STATE_SHOW = TMW.CONST.STATE.DEFAULT_SHOW
+
+Type:RegisterConfigPanel_XMLTemplate(165, "TellMeWhen_IconStates", {
+	[STATE_SHOW]     = { order = 1, text = "|cFF00FF00" .. L["DEFAULT"], },
 })
 
-local function Value_OnUpdate(icon)
-	if not icon.script_values.triggerFunc(icon) then
-		return
-	end
 
-	if icon.script_values.changed then
-		icon:YieldInfo(true, icon.script_values.current, icon.script_values.max, icon.script_values.colors)
+local function Value_OnUpdate(icon)
```

Comment:
A number of changes in here:

- Upvalued script_values for readability
- Changed to just straight `:SetInfo()` calls instead of routing through `:YieldInfo()` and `:HandleYieldedInfo()`, which are used for group controller icons.
- Added STATE_SHOW as the state that gets set. If `triggerFunc` is meant to hide the icon when false, you can do `icon:SetInfo("state", 0)`, or if you want, introduce `TMW.CONST.STATE.DEFAULT_HIDE` for hidden and use that state instead of 0.